src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = Object.assign({}, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
```

Comment:
As for me, such cases are easier to read when they are divided.
```suggestion
        break;

      case 'removeProperties':
```

Reply:
There are indentations of 2 spaces between break and next case